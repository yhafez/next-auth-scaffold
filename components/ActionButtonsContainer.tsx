// Path: ./components/ActionButtonsContainer.tsx
import { ReactNode } from 'react'
import { Box } from '@mui/material'

export interface ActionButtonsContainerProps {
	name: string
	children: ReactNode
}

export default function ActionButtonsContainer({ name, children }: ActionButtonsContainerProps) {
	return (
		<Box
			id={`${name}-buttons-container`}
			sx={{
				display: 'flex',
				flexDirection: 'row',
				justifyContent: 'space-around',
				alignItems: 'center',
				gap: 2,
				margin: '8px 4px',
				width: '80%',
			}}
		>
			{children}
		</Box>
	)
}
