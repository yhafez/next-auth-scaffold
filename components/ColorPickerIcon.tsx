// Path: ./components/ColorPicker.tsx
import { useState, MouseEvent, useCallback, useEffect } from 'react'
import { Box, IconButton, Typography } from '@mui/material'
import { ColorLens } from '@mui/icons-material'
import { darken, lighten } from 'color2k'

import { ColorPickerPopover } from '.'
import { useBoundStore } from '../store'
import {
	getContrastColor,
	getSecondaryColor,
	mixColors,
	findOptimalOverlayOpacity,
	isOverlayNecessary,
} from '../utils/helpers'

export interface ColorPickerProps {
	name: string
	selectedInit?: boolean
}

export default function ColorPickerIcon({ name, selectedInit }: ColorPickerProps) {
	const { darkMode, customPalette, setCustomPalette } = useBoundStore()

	const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null)
	const [selected, setSelected] = useState(selectedInit ?? false)

	const handleColorChange = useCallback((color: string) => {
		const contrast = getContrastColor(color)
		let mixedColorPrimaryMain = color
		if (isOverlayNecessary(color, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(contrast, overlayColor, color, 7)
			mixedColorPrimaryMain = mixColors(color, overlayColor, overlayOpacity)
		}
		let mixedColorPrimaryLight = lighten(mixedColorPrimaryMain, 0.2)
		if (isOverlayNecessary(mixedColorPrimaryLight, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				contrast,
				overlayColor,
				mixedColorPrimaryLight,
				7,
			)
			mixedColorPrimaryLight = mixColors(mixedColorPrimaryLight, overlayColor, overlayOpacity)
		}
		let mixedColorPrimaryDark = darken(mixedColorPrimaryMain, 0.2)
		if (isOverlayNecessary(mixedColorPrimaryDark, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				contrast,
				overlayColor,
				mixedColorPrimaryDark,
				7,
			)
			mixedColorPrimaryDark = mixColors(mixedColorPrimaryDark, overlayColor, overlayOpacity)
		}

		const secondaryColor = getSecondaryColor(mixedColorPrimaryMain)
		const secondaryColorContrast = getContrastColor(secondaryColor)
		let mixedColorSecondaryMain = secondaryColor
		if (isOverlayNecessary(secondaryColor, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				secondaryColor,
				7,
			)
			mixedColorSecondaryMain = mixColors(secondaryColor, overlayColor, overlayOpacity)
		}
		let mixedColorSecondaryLight = lighten(mixedColorSecondaryMain, 0.2)
		if (isOverlayNecessary(mixedColorSecondaryLight, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				mixedColorSecondaryLight,
				7,
			)
			mixedColorSecondaryLight = mixColors(mixedColorSecondaryLight, overlayColor, overlayOpacity)
		}
		let mixedColorSecondaryDark = darken(mixedColorSecondaryMain, 0.2)
		if (isOverlayNecessary(mixedColorSecondaryDark, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				mixedColorSecondaryDark,
				7,
			)
			mixedColorSecondaryDark = mixColors(mixedColorSecondaryDark, overlayColor, overlayOpacity)
		}

		const newPalette = {
			...customPalette,
			primary: {
				main: mixedColorPrimaryMain,
				contrastText: contrast,
				light: mixedColorPrimaryLight,
				dark: mixedColorPrimaryDark,
			},
			secondary: {
				main: mixedColorSecondaryMain,
				contrastText: secondaryColorContrast,
				light: mixedColorSecondaryLight,
				dark: mixedColorSecondaryDark,
			},
		}

		setCustomPalette(newPalette)
	}, [])

	useEffect(() => {
		const customPalette = localStorage.getItem('customPalette')
		if (customPalette) handleColorChange(customPalette)
	}, [handleColorChange])

	useEffect(() => {
		setSelected(selectedInit ?? false)
	}, [selectedInit])

	const open = Boolean(anchorEl)
	const id = open ? 'simple-popover' : undefined

	const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
		setAnchorEl(event.currentTarget)
	}

	const handleClose = () => {
		setSelected(false)
		setAnchorEl(null)
	}

	return (
		<>
			<Box
				id={`${name}-color-picker-container`}
				sx={{
					display: 'flex',
					flexDirection: 'column',
					alignItems: 'center',
					justifyContent: 'center',
					p: 0,
					height: '100%',
					width: '100%',
					cursor: 'pointer',
				}}
			>
				<IconButton
					id={`${name}-color-picker-button`}
					aria-describedby={id}
					aria-labelledby={`${name}-color-picker-label`}
					onClick={handleClick}
				>
					<ColorLens
						id={`${name}-color-picker-icon`}
						sx={{
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
							fontSize: '1.5rem',
						}}
						aria-labelledby={`${name}-color-picker-label`}
						role="img"
					/>
				</IconButton>
				<label htmlFor={`${name}-color-picker-button`}>
					<Typography
						id={`${name}-color-picker-label`}
						variant="body2"
						sx={{
							textAlign: 'center',
							fontWeight: 500,
							cursor: 'pointer',
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					>
						Theme Color
					</Typography>
				</label>
			</Box>
			<ColorPickerPopover
				name={name}
				anchorEl={anchorEl}
				handleClose={handleClose}
				anchorOrigin={{
					vertical: 'bottom',
					horizontal: 'center',
				}}
				transformOrigin={{
					vertical: 'top',
					horizontal: 'center',
				}}
				handleColorChange={handleColorChange}
			/>
		</>
	)
}
