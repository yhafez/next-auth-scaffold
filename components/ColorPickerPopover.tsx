// Path: ./components/ColorPickerPopover.tsx

import { useState, useCallback } from 'react'
import { Popover, PopoverOrigin } from '@mui/material'
import { useBoundStore } from '../store'
import { ChromePicker } from 'react-color'

export interface ColorPickerPopoverProps {
	name: string
	open: boolean
	anchorEl: HTMLButtonElement | null
	handleClose: () => void
	handleColorChange: (color: string) => void
	anchorOrigin?: PopoverOrigin
	transformOrigin?: PopoverOrigin
}

const ColorPickerPopover = ({
	name,
	open,
	anchorEl,
	handleClose,
	handleColorChange,
	anchorOrigin,
	transformOrigin,
}: ColorPickerPopoverProps) => {
	const { customPalette } = useBoundStore()

	const [color, setColor] = useState(customPalette.primary.main)

	const handleChange = useCallback(
		(color: any) => {
			setColor(color.hex)
			handleColorChange(color.hex)
		},
		[handleColorChange],
	)

	return (
		<Popover
			id={`${name}-color-picker-popover`}
			open={open}
			anchorEl={anchorEl}
			onClose={handleClose}
			anchorOrigin={anchorOrigin}
			transformOrigin={transformOrigin}
			sx={{
				cursor: 'pointer',
				'& .MuiPopover-paper': {
					padding: 2,
					backgroundColor: 'background.paper',
					color: 'text.primary',
				},
			}}
		>
			<ChromePicker color={color} onChange={color => handleChange(color)} disableAlpha />
		</Popover>
	)
}

export default ColorPickerPopover
