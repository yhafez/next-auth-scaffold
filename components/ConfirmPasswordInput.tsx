// Path: ./components/ConfirmPasswordInput.tsx
import { Dispatch, SetStateAction, KeyboardEvent } from 'react'

import PasswordInput from './PasswordInput'

export interface ConfirmPasswordInputProps {
	name: string
	password: string
	setPassword: Dispatch<SetStateAction<string>>
	confirmPassword: string
	setConfirmPassword: Dispatch<SetStateAction<string>>
	disabled?: boolean
	handleEnter?: (e: KeyboardEvent) => void
}

export default function ConfirmPasswordInput({
	name,
	password,
	setPassword,
	confirmPassword,
	setConfirmPassword,
	disabled = false,
	handleEnter,
}: ConfirmPasswordInputProps) {
	return (
		<>
			<PasswordInput
				name={name}
				value={password}
				setValue={setPassword}
				disabled={disabled}
				handleEnter={handleEnter}
			/>
			<PasswordInput
				name={`${name}`}
				value={confirmPassword}
				setValue={setConfirmPassword}
				disabled={disabled}
				isConfirmPassword={true}
				handleEnter={handleEnter}
			/>
		</>
	)
}
