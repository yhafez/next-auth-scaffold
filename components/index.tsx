// Path: ./components/index.tsx
import Layout from './Layout'

import DashboardDrawer from './Drawers/DashboardDrawer'
import SettingsDrawer from './Drawers/SettingsDrawer'

import ColorPickerPopover from './Popovers/ColorPickerPopover'
import DrawerSignOutPopover from './Popovers/DrawerSignOutPopover'

import ActionButtonsContainer from './ActionButtonsContainer'
import ColorPickerIcon from './ColorPickerIcon'
import ConfirmPasswordInput from './ConfirmPasswordInput'
import DarkModeSwitch from './DarkModeSwitch'
import Drawer from './Drawer'
import DrawerButton from './DrawerButton'
import EditableTextField from './EditableTextField'
import EmailInput from './EmailInput'
import Modal from './Modal'
import ModalButton from './ModalButton'
import ModalNote from './ModalNote'
import NavigationButton from './NavigationButton'
import PasswordInput from './PasswordInput'
import SubmitButton from './SubmitButton'
import VisibilityIconButton from './VisibilityIconButton'

export {
	Layout,
	DashboardDrawer,
	SettingsDrawer,
	ColorPickerPopover,
	DrawerSignOutPopover,
	ActionButtonsContainer,
	ColorPickerIcon,
	ConfirmPasswordInput,
	DarkModeSwitch,
	Drawer,
	DrawerButton,
	EditableTextField,
	EmailInput,
	Modal,
	ModalButton,
	ModalNote,
	NavigationButton,
	PasswordInput,
	SubmitButton,
	VisibilityIconButton,
}
