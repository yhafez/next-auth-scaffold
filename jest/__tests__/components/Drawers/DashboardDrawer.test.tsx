// Path: ./jest/__tests__/components/Drawers/DashboardDrawer.test.tsx

import { render, act, screen, waitFor } from '@testing-library/react'
import { axe } from 'jest-axe'

import { DashboardDrawer } from '../../../../components'

describe('DashboardDrawer', () => {
	let container: HTMLElement

	beforeEach(() => {
		act(() => {
			container = render(
				<main>
					<DashboardDrawer />
				</main>,
			).container

			return container
		})
	})

	it('Should have no accessibility violations', async () => {
		waitFor(async () => expect(await axe(screen.getByRole('main'))).toHaveNoViolations())
	})

	it('Should have a Box element with id "dashboard-drawer"', () => {
		const test = container.querySelector('#dashboard-drawer')
		expect(test).toBeTruthy()
	})

	it('Should have a Box element with id "dashboard-drawer-header"', () => {
		const test = container.querySelector('#dashboard-drawer-header')
		expect(test).toBeTruthy()
	})

	it('Should have a Box element with id "dashboard-drawer-header-user"', () => {
		const test = container.querySelector('#dashboard-drawer-header-user')
		expect(test).toBeTruthy()
	})

	it('Should have a Button element with id "dashboard-drawer-user-avatar"', () => {
		const test = container.querySelector('#dashboard-drawer-user-avatar')
		expect(test).toBeTruthy()
	})

	it('Should have an Avatar element with id "dashboard-drawer-user-avatar-image"', () => {
		const test = container.querySelector('#dashboard-drawer-user-avatar-image')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with id "dashboard-drawer-user-avatar-name"', () => {
		const test = container.querySelector('#dashboard-drawer-user-avatar-name')
		expect(test).toBeTruthy()
	})

	it('Should have a List element with id "dashboard-drawer-header-actions"', () => {
		const test = container.querySelector('#dashboard-drawer-header-actions')
		expect(test).toBeTruthy()
	})

	it('Should have a ListItem element with id "dashboard-drawer-header-actions-home-list-item"', () => {
		const test = container.querySelector('#dashboard-drawer-header-actions-home-list-item')
		expect(test).toBeTruthy()
	})

	it('Should have a Link element with id "dashboard-home-drawer-link"', () => {
		const test = container.querySelector('#dashboard-home-drawer-link')
		expect(test).toBeTruthy()
	})

	it('Should have an Icon element with id "dashboard-home-drawer-icon"', () => {
		const test = container.querySelector('#dashboard-home-drawer-icon')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with id "dashboard-home-drawer-button-label"', () => {
		const test = container.querySelector('#dashboard-home-drawer-button-label')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "Home"', () => {
		const test = container.querySelector('#dashboard-home-drawer-button-label')?.textContent
		expect(test).toBe('Home')
	})

	it('Should have a ListItem element with id "dashboard-drawer-header-actions-profile-list-item"', () => {
		const test = container.querySelector('#dashboard-drawer-header-actions-profile-list-item')
		expect(test).toBeTruthy()
	})

	it('Should have a Link element with id "dashboard-profile-drawer-link"', () => {
		const test = container.querySelector('#dashboard-profile-drawer-link')
		expect(test).toBeTruthy()
	})

	it('Should have an Icon element with id "dashboard-profile-drawer-icon"', () => {
		const test = container.querySelector('#dashboard-profile-drawer-icon')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with id "dashboard-profile-drawer-button-label"', () => {
		const test = container.querySelector('#dashboard-profile-drawer-button-label')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "Profile"', () => {
		const test = container.querySelector('#dashboard-profile-drawer-button-label')?.textContent
		expect(test).toBe('Profile')
	})

	it('Should have a ListItem element with id "dashboard-drawer-header-actions-settings-list-item"', () => {
		const test = container.querySelector('#dashboard-drawer-header-actions-settings-list-item')
		expect(test).toBeTruthy()
	})

	it('Should have a Link element with id "dashboard-settings-drawer-link"', () => {
		const test = container.querySelector('#dashboard-settings-drawer-link')
		expect(test).toBeTruthy()
	})

	it('Should have an Icon element with id "dashboard-settings-drawer-icon"', () => {
		const test = container.querySelector('#dashboard-settings-drawer-icon')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with id "dashboard-settings-drawer-button-label"', () => {
		const test = container.querySelector('#dashboard-settings-drawer-button-label')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "Settings"', () => {
		const test = container.querySelector('#dashboard-settings-drawer-button-label')?.textContent
		expect(test).toBe('Settings')
	})

	it('Should have a List element with id "dashboard-drawer-footer"', () => {
		const test = container.querySelector('#dashboard-drawer-footer')
		expect(test).toBeTruthy()
	})

	it('Should have a ListItem element with id "dashboard-drawer-footer-color-picker-list-item"', () => {
		const test = container.querySelector('#dashboard-drawer-footer-color-picker-list-item')
		expect(test).toBeTruthy()
	})

	it('Should have an IconButton element with id "dashboard-color-picker-drawer-button"', () => {
		const test = container.querySelector('#dashboard-color-picker-drawer-button')
		expect(test).toBeTruthy()
	})

	it('Should have an Icon element with id "dashboard-color-picker-drawer-icon"', () => {
		const test = container.querySelector('#dashboard-color-picker-drawer-icon')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with id "dashboard-color-picker-drawer-button-label"', () => {
		const test = container.querySelector('#dashboard-color-picker-drawer-button-label')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "Theme Color"', () => {
		const test = container.querySelector('#dashboard-color-picker-drawer-button-label')?.textContent
		expect(test).toBe('Theme Color')
	})

	it('Should have a ListItem element with id "dashboard-drawer-footer-dark-mode-list-item"', () => {
		const test = container.querySelector('#dashboard-drawer-footer-dark-mode-list-item')
		expect(test).toBeTruthy()
	})

	it('Should have an IconButton element with id "dashboard-dark-mode-drawer-button"', () => {
		const test = container.querySelector('#dashboard-dark-mode-drawer-button')
		expect(test).toBeTruthy()
	})

	it('Should have an Icon element with id "dashboard-dark-mode-drawer-icon"', () => {
		const test = container.querySelector('#dashboard-dark-mode-drawer-icon')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with id "dashboard-dark-mode-drawer-button-label"', () => {
		const test = container.querySelector('#dashboard-dark-mode-drawer-button-label')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "Dark Mode" or "Light Mode"', () => {
		const test = container.querySelector('#dashboard-dark-mode-drawer-button-label')?.textContent
		const options = ['Dark Mode', 'Light Mode']
		expect(options).toContain(test)
	})

	it('Should have a ListItem element with id "dashboard-drawer-footer-log-out-list-item"', () => {
		const test = container.querySelector('#dashboard-drawer-footer-log-out-list-item')
		expect(test).toBeTruthy()
	})

	it('Should have an IconButton element with id "dashboard-log-out-drawer-button"', () => {
		const test = container.querySelector('#dashboard-log-out-drawer-button')
		expect(test).toBeTruthy()
	})

	it('Should have an Icon element with id "dashboard-log-out-drawer-icon"', () => {
		const test = container.querySelector('#dashboard-log-out-drawer-icon')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with id "dashboard-log-out-drawer-button-label"', () => {
		const test = container.querySelector('#dashboard-log-out-drawer-button-label')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "Logout"', () => {
		const test = container.querySelector('#dashboard-log-out-drawer-button-label')?.textContent
		expect(test).toBe('Logout')
	})

	it('Should math snapshot', () => {
		expect(container).toMatchSnapshot()
	})
})
