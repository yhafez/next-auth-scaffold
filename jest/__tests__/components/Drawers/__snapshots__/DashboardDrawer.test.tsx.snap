// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardDrawer Should math snapshot 1`] = `
<div>
  <main>
    <nav
      aria-label="left-hand drawer dashboard navigation"
      class="MuiBox-root css-r0x2ub"
      id="dashboard-drawer"
    >
      <div
        class="MuiBox-root css-1hnpd1e"
        id="dashboard-drawer-header"
      >
        <div
          class="MuiBox-root css-1vwnqvu"
          id="dashboard-drawer-header-user"
        >
          <label
            for="avatar"
            style="width: 100%;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation css-186bged-MuiButtonBase-root-MuiButton-root"
              id="dashboard-drawer-user-avatar"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-1oin0wn-MuiAvatar-root"
                id="dashboard-drawer-user-avatar-image"
              >
                <img
                  alt="User's avatar"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="/default_profile.svg"
                />
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 css-1md76xf-MuiTypography-root"
                id="dashboard-drawer-user-avatar-name"
              >
                User
              </p>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </label>
        </div>
        <ul
          class="MuiList-root MuiList-padding css-16p2v77-MuiList-root"
          id="dashboard-drawer-header-actions"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ykqd4v-MuiListItem-root"
            id="dashboard-drawer-header-actions-home-list-item"
          >
            <a
              aria-labelledby="dashboard-home-drawer-button-label"
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-hwqx0i-MuiTypography-root-MuiLink-root"
              href="/"
              id="dashboard-home-drawer-link"
              title="Home"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-b6p50a-MuiTypography-root"
                id="dashboard-home-drawer-button-label"
              >
                Home
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                id="dashboard-home-drawer-icon"
                role="img"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </a>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ykqd4v-MuiListItem-root"
            id="dashboard-drawer-header-actions-profile-list-item"
          >
            <a
              aria-labelledby="dashboard-profile-drawer-button-label"
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-hwqx0i-MuiTypography-root-MuiLink-root"
              href="/profile"
              id="dashboard-profile-drawer-link"
              title="Profile"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-b6p50a-MuiTypography-root"
                id="dashboard-profile-drawer-button-label"
              >
                Profile
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                id="dashboard-profile-drawer-icon"
                role="img"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
                />
              </svg>
            </a>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ykqd4v-MuiListItem-root"
            id="dashboard-drawer-header-actions-settings-list-item"
          >
            <a
              aria-labelledby="dashboard-settings-drawer-button-label"
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-hwqx0i-MuiTypography-root-MuiLink-root"
              href="/settings"
              id="dashboard-settings-drawer-link"
              title="Settings"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-b6p50a-MuiTypography-root"
                id="dashboard-settings-drawer-button-label"
              >
                Settings
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SettingsIcon"
                focusable="false"
                id="dashboard-settings-drawer-icon"
                role="img"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <ul
        class="MuiList-root MuiList-padding css-1f4h1di-MuiList-root"
        id="dashboard-drawer-footer"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ykqd4v-MuiListItem-root"
          id="dashboard-drawer-footer-color-picker-list-item"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-tsm0ra-MuiButtonBase-root-MuiIconButton-root"
            id="dashboard-color-picker-drawer-button"
            tabindex="0"
            type="button"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-b6p50a-MuiTypography-root"
              id="dashboard-color-picker-drawer-button-label"
            >
              Theme Color
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PaletteIcon"
              focusable="false"
              id="dashboard-color-picker-drawer-icon"
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.49 2 2 6.49 2 12s4.49 10 10 10c1.38 0 2.5-1.12 2.5-2.5 0-.61-.23-1.2-.64-1.67-.08-.1-.13-.21-.13-.33 0-.28.22-.5.5-.5H16c3.31 0 6-2.69 6-6 0-4.96-4.49-9-10-9zm5.5 11c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm-3-4c-.83 0-1.5-.67-1.5-1.5S13.67 6 14.5 6s1.5.67 1.5 1.5S15.33 9 14.5 9zM5 11.5c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5S7.33 13 6.5 13 5 12.33 5 11.5zm6-4c0 .83-.67 1.5-1.5 1.5S8 8.33 8 7.5 8.67 6 9.5 6s1.5.67 1.5 1.5z"
              />
            </svg>
          </button>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ykqd4v-MuiListItem-root"
          id="dashboard-drawer-footer-dark-mode-list-item"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-tsm0ra-MuiButtonBase-root-MuiIconButton-root"
            id="dashboard-dark-mode-drawer-button"
            tabindex="0"
            type="button"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-b6p50a-MuiTypography-root"
              id="dashboard-dark-mode-drawer-button-label"
            >
              Dark Mode
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="Brightness4Icon"
              focusable="false"
              id="dashboard-dark-mode-drawer-icon"
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
              />
            </svg>
          </button>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ykqd4v-MuiListItem-root"
          id="dashboard-drawer-footer-log-out-list-item"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-tsm0ra-MuiButtonBase-root-MuiIconButton-root"
            id="dashboard-log-out-drawer-button"
            tabindex="0"
            type="button"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-b6p50a-MuiTypography-root"
              id="dashboard-log-out-drawer-button-label"
            >
              Logout
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExitToAppIcon"
              focusable="false"
              id="dashboard-log-out-drawer-icon"
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M10.09 15.59 11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </main>
</div>
`;
